### PR TITLE
fixup! Validate .spacemacs variables.

### PR DESCRIPTION
**Pull request by lebensterben**
fixes syl20bnr#14535 syl20bnr#14537 syl20bnr#14529 syl20bnr#10638 (comment)

### CODE REVIEW

**Comment by lebensterben** [comment]
@JAremko Here we go

File: core/core-dotspacemacs.el [hunk 1]
```diff
@@ -171,7 +171,7 @@ a layer lazily."
 wrapped in a layer. If you need some configuration for these
 packages then consider to create a layer, you can also put the
 configuration in `dotspacemacs/user-config'."
-  '(repeat symbol)
+  '(repeat (choice symbol (cons symbol sexp)))
```
Comment (JAremko) [inline core/core-dotspacemacs.el:174]: maybe the type can be stricter here.

In the case:

```
   dotspacemacs-additional-packages '(
     amx
     (beacon :location (recipe :fetcher github :repo "Malabarba/beacon")))
```
it looks like in the cons case we always have first 2 elements as symbols.  So maybe '(repeat (choice symbol (cons symbol (const symbol sexp))) will work :thinking:

Comment (JAremko) [reply core/core-dotspacemacs.el:174]: On the other hands it's too cryptic.

**Comment by JAremko** [comment]
It's good as it is. Thanks agan.

**Pull request merged**